buildermgr/common.go

```diff
@@ -94,9 +94,15 @@ func buildPackage(fissionClient *crd.FissionClient, envBuilderNamespace string,
 
 	log.Printf("Build succeed, source package: %v, deployment package: %v", srcPkgFilename, buildResp.ArtifactFilename)
 
+	archivePackage := true
+	if env.Spec.KeepArchive {
+		archivePackage = !env.Spec.KeepArchive
+	}
```

Comment:
Not sure if you saw my previous comment: save yourself 3 lines with `archivePackage := !env.Spec.KeepArchive`